### PR TITLE
chore(issue/pr_template): rework and modernize templates

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pull request

- New Issue templates
- New PR template
- Icons for links in the Issue choose menu

New Issue selection + icons for each link on issue/choose:
image

Furthermore this PR reassembles the new PR template

### Issues Addressed:

- None
### Tests Performed:

- Created some issues on the fork
- Visual Tests regarding the issues
- Labels are created successfully
- PR visual test via preview

### How to Test the Changes:

Go to [my fork](https://github.com/ulfgebhardt/azerothcore-wotlk/issues) and [open an Issue](https://github.com/ulfgebhardt/azerothcore-wotlk/issues/new/choose)

You can see an blank issue opened for each issue type [there](https://github.com/ulfgebhardt/azerothcore-wotlk/issues):

image

### Known Issues and TODO List:

- [ ] Labels need to be created if they should be used. Those labels are `bug`, `feature`, `npc`, `spell`, `quest` and `data`. Those are optional but bring the benefit of auto labeling.

---

| When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**. |
| --- |

| You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions [here](http://www.azerothcore.org/wiki/How-to-test-a-PR) |
| --- |

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (FrancescoBorzi) [inline .github/ISSUE_TEMPLATE/bug_report_spell.md:96]: is there a way to avoid this repetition? It can became an issue since we will then have to change the same thing in multiple places. Maybe we can stick with 1 bug report category and let the user specify what's about inside the template.

Apart from this, good job overall. I like the improvements

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (FrancescoBorzi) [inline .github/PULL_REQUEST_TEMPLATE.md:6]: Pull Request

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (FrancescoBorzi) [inline .github/PULL_REQUEST_TEMPLATE.md:47]: this is nice, I didn't know about it

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (ulfgebhardt) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:96]: I understand your question that you do not want to have the duplicate content in the different bug report files.

You can not use any sort of scripting inside the issue templates to my knowledge, nor is it possible to to combine template snippets. Therefore the only way would be to blow up the templates to include all possible variations like so:

```
<!--- uncomment the following if you want to report a spell bug --->
<!---
Some spell specific stuff
--->
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (ulfgebhardt) [reply .github/PULL_REQUEST_TEMPLATE.md:47]: corrected

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (ulfgebhardt) [reply .github/PULL_REQUEST_TEMPLATE.md]: corrected

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (FrancescoBorzi) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:96]: unfortunately, I'm not aware of any scripting, so I think the best thing would be to use templates like you just suggested

**Review by Helias — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (Kitzunu) [inline .github/ISSUE_TEMPLATE/bug_report_spell.md:5]: Leaving my two cents
1. Remove the bug label and title
2. Personally I am a fan of the DB/Spell: Core/Spell etc naming for all cases because i personally think it gives a cleaner look.

File: .github/ISSUE_TEMPLATE/feature_request.md
Comment (Kitzunu) [inline .github/ISSUE_TEMPLATE/feature_request.md:16]: This really doesnt make sense to make drawings, better to keep tell people to explain, and if they choose they could use images but to strictly limit to images, no.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (Kitzunu) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: Dont remove

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (Kitzunu) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: Dont remove

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (Kitzunu) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: Personally I prefer to have tests performed before how to test to so it is in order
1. What has been done
2. What needs to be done

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (Kitzunu) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:5]: Or if we create a label for the two, "spell" is just too generic, same goes for crestures

File: .github/ISSUE_TEMPLATE/bug_report_data.md
Comment (Kitzunu) [inline .github/ISSUE_TEMPLATE/bug_report_data.md:2]: Explain this please, because it doesn' make amy sense to me

**Comment by Kitzunu** [comment]
Looks okay, just look over my reviews

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (ulfgebhardt) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:5]: Thats the whole point of this - preordering, giving visual feedback on the issue list by default

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (ulfgebhardt) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done

File: .github/ISSUE_TEMPLATE/feature_request.md
Comment (ulfgebhardt) [reply .github/ISSUE_TEMPLATE/feature_request.md:16]: This section  was unified under the feature itself:

```
## 🚀 Feature
<!-- Describe your feature request or suggestion in detail. A clear and concise description of what you want to happen. -->


[...]
```

Naturally when you submit a feature request you propose a solution and I believe its not necessary to split into two sections as it was before:
- Describe your feature request or suggestion in detail
- Describe a possible solution to your Feature or Suggestion in detail

Furthermore I believe that encouraging people do draw something about their feature will make them think about how to explain it in a way that more people can understand it. Having a visualization can help to make people understand
> A picture is worth a thousand words
There is this saying for a reason.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (ulfgebhardt) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: I did not remove it. It is replaced with

```
### Issues Addressed:
<!-- If the issue doesn't exist, describe it and how to reproduce it, please. If the issue already exists, just paste the link to the issue you close, like this: Closes https://github.com/azerothcore/azerothcore-wotlk/issues/967 -->
- None
```

And while `closes` is the keyword to autoclose the issue I think experienced developers tend to know that - besides there is alternatives like `fixes`. I analysed some PRs of yours and most just have a stray `Closes` in them while `None` is more precise in 95% of all cases.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (ulfgebhardt) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: This section is unified under:

```
## 🍰 Pull request
<!-- Describe the Pull request. What changes do you propose? -->

[...]
```
A similar Issue like with the feature request - If you do a PR you ted to change things, thats the point of the PR - naturally a developer describes what he does in one paragraph.

Compare:

image
image

File: .github/ISSUE_TEMPLATE/bug_report_data.md
Comment (ulfgebhardt) [reply .github/ISSUE_TEMPLATE/bug_report_data.md:2]: This is from what I lately learned at `vmangos` - the main type or issue or PR is about incorrect or missing data which is not aligned with the sniffs or what ever. It might be a bit different with WOTLK since there is only old sniff data - I just assume that the main Issues you face are still broken quests, incorrect data or simply missing data and not bugs with the core or subsystems itself.

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (ulfgebhardt) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:5]: And here is a visual comparison of the outome:

image

image

As clearly can be seen not everyone follows the guidlines, especially since the system is new at `vmangos` while others fail to name their issue at all.

In the long run you will have everyone comply tho as seen at `cmangos` (I did not implement bugreport subtypes there so its a bit uniform)

image

File: .github/ISSUE_TEMPLATE/bug_report_data.md
Comment (Kitzunu) [reply .github/ISSUE_TEMPLATE/bug_report_data.md:2]: I mean we dont have sniffs and personally my first thoyght was that anything can be missing data, for example if a crrsture is missing immunities, is it a missing data or a cresture bug. Well both.

So I dont know :shrug:

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (Kitzunu) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: Add back at lesst one bulletpoint underneathe the section so it is more clear <3

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (Kitzunu) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: Personally i prefered to have Closes there. Anyone else got any thoughts?

File: .github/ISSUE_TEMPLATE/feature_request.md
Comment (Kitzunu) [reply .github/ISSUE_TEMPLATE/feature_request.md:16]: I disagree, if I am going to make s festure im not going to draw ehat I mean. For example if I want the festure to be able to teleport to instance ID there is no way to drsw that.

I get your point but believe it is not useful in wow emulation

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (Kitzunu) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:5]: Sure but [bug] is unnecesssry and my point still stands.

I would prefer we manually lable them instead in that case because most of the issues you get in any category are issues in either the DB or Core/Script so if you f.ex only label it as "spell" it will be 1000 times harded for s new contributer to choose an issue incase they only know cpp or db.

In this case generalizing is not good. Neither is adding more issue templates.

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (FrancescoBorzi) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:96]: @ulfgebhardt maybe I was not clear, but I meant this:

> Therefore the only way would be to blow up the templates to include all possible variations like so:

to avoid duplicating the same template among different `bug_*` files. So maybe it's better to have only one bug template  file and use the template to include variations (inside it)

File: .github/ISSUE_TEMPLATE/feature_request.md
Comment (FrancescoBorzi) [reply .github/ISSUE_TEMPLATE/feature_request.md:16]: I think we can just have it as an optional field. For example:

`### Visualization (Optional)`

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (FrancescoBorzi) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:5]: I'm also against having multiple issue templates, as I pointed in my other comment.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (FrancescoBorzi) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: I think both "Changes proposed" or "Pull request" can work fine, as long as it's clear to the user what he/she needs to write there

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (FrancescoBorzi) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: I agree with `None`. People should read the template and modify it accordingly. Some people don't do it and leave it "Closes" but it's their fault, not ours.

File: .github/ISSUE_TEMPLATE/feature_request.md
Comment (ulfgebhardt) [reply .github/ISSUE_TEMPLATE/feature_request.md:16]: I don't think that's smart, but done

File: .github/ISSUE_TEMPLATE/bug_report_data.md
Comment (ulfgebhardt) [reply .github/ISSUE_TEMPLATE/bug_report_data.md:2]: As stated above you should find a few very general types of bugreports users can use to precategorize them and give maintainers a visual feedback - my experience tells me that data related stuff is a very common on

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (ulfgebhardt) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:96]: I disagree

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (ulfgebhardt) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:5]: I disagree

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (FrancescoBorzi) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:96]: @ulfgebhardt I guess we can give it a try, as long as you promise to take care of keeping all the templates in synch :)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (Kitzunu) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: > Add back at lesst one bulletpoint underneathe the section so it is more clear <3

File: .github/ISSUE_TEMPLATE/bug_report_spell.md
Comment (Kitzunu) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:5]: First of all, stop resolving this when it is not resolved.

Second of all, why do you disagree? Generalizing is bad, especially for new contributors. Now there is no easy solution for that except for good old manual labeling, so that's why I don't believe in making them auto label with "creature" "spell" etc. For the actual template, I have no issue with it.

Also, again, labeling with "bug" is very unnecessary especially since an open issue does not mean it is a bug, a lot of open reports are just due to outdated cores and therefore not a bug.

Comment (FrancescoBorzi) [reply .github/ISSUE_TEMPLATE/bug_report_spell.md:5]: Yes, please don't mark a discussion as resolved until it's actually resolved. BTW I marked the other discussion as resolved so we can just talk here. As I just said:

@ulfgebhardt I guess we can give it a try, as long as you promise to take care of keeping all the templates in synch :)

so we can give it a try and see how it goes

**Comment by Kitzunu** [comment]
I honestly believe we should merge this once we work through the small things.

**Comment by FrancescoBorzi** [comment]
I agree, not sure why @ulfgebhardt has closed it

**Comment by FrancescoBorzi** [comment]
closing as apparently there is no will to keep working on this for the time being

@ulfgebhardt feel free to reopen if you'll ever want to go back on this kind of work.

**Pull request closed without merging**
